test/test.js

```diff
@@ -458,6 +458,19 @@ function checkRefTestResults(browser, id, results) {
   });
 }
 
+function browserTestReportHandler(req,res) {
+  var parsedUrl = url.parse(req.url, true);
+  var pathname = parsedUrl.pathname;
+  if (pathname === '/browserTestReports') {
+    var information = req.body.browserTestInfo;
```

Comment:
This `req.body` does not exist in the Node.js API.
To read the POST data (whatever you pass to the `send` method of `XMLHttpRequest`), you need to collect all data chunks and combine them. The "data" event will be emitted until there is no more data, and then the "end" event is emitted, upon which you can use all collected chunks and write it to a file. Instead of buffering all data, in this specific case you can also directly write the data to a file:

- Use [`fs.createWriteStream`](https://nodejs.org/api/fs.html#fs_fs_createwritestream_path_options) to create a writable stream.
- In the "data" event of `req`, call the `write` method on the writable stream.
- In the "end" event of `req`, call the `end` method on the writable stream.

Here is an example from Node.js's core documentation: https://nodejs.org/api/stream.html#stream_writable_end_chunk_encoding_callback
Note that the chunks that you receive from the "data" event are Buffer objects instead of strings, but that is totally fine since the `write` method also accepts Buffers (as you can see from the documentation).